azurelinuxagent/ga/collect_logs.py

```diff
@@ -170,29 +175,33 @@ def _collect_logs(self):
         # Invoke the command line tool in the agent to collect logs, with resource limits on CPU and memory (RAM).
 
         systemd_cmd = [
-            "systemd-run", "--unit={0}".format(logcollector.CGROUPS_UNIT),
+            "systemd-run", "--property=CPUAccounting=no", "--property=MemoryAccounting=no",
+            "--unit={0}".format(logcollector.CGROUPS_UNIT),
             "--slice={0}".format(cgroupconfigurator.LOGCOLLECTOR_SLICE), "--scope"
         ]
 
         # The log tool is invoked from the current agent's egg with the command line option
         collect_logs_cmd = [sys.executable, "-u", sys.argv[0], "-collect-logs"]
         final_command = systemd_cmd + collect_logs_cmd
 
-        def exec_command(output_file):
+        def exec_command():
             start_time = datetime.datetime.utcnow()
             success = False
             msg = None
             try:
-                shellutil.run_command(final_command, log_error=False, stdout=output_file, stderr=output_file)
-                duration = elapsed_milliseconds(start_time)
-                archive_size = os.path.getsize(COMPRESSED_ARCHIVE_PATH)
-
-                msg = "Successfully collected logs. Archive size: {0} b, elapsed time: {1} ms.".format(archive_size,
-                                                                                                    duration)
-                logger.info(msg)
-                success = True
-
-                return True
+                base_dir = conf.get_lib_dir()
```

Comment:
Since the main logger set to capture stdout,  systemd-run stdout to temp file instead of agent log file to avoid duplicates in agent log

Reply:
could you explain this comment? thanks

Reply:
Duplicate logs in agent log are due to `systemd-run popen `uses the agent log file for stdout to write and also, agent initialized the logger to capture the any stdout. Since both are using same log file, we would see duplicates. So, I changed the popen stdout to use standard pipe instead of log file, so that popen won't write to the file.

Reply:
sorry, still don't get it. I think this comment is misplaced, can't find the corresponding code, can you point me to it? thanks

Reply:
this line https://github.com/Azure/WALinuxAgent/pull/2637/files#diff-35808f273042b1e0b61f6e439a3469a775aadb1e4850db80c742480435fb6258R193